cli/tools/test.rs

```diff
@@ -508,67 +522,80 @@ pub async fn run_tests(
       .await?;
   }
 
-  let prepare_roots = {
-    let mut files = Vec::new();
-    let mut fetch_permissions = Permissions::allow_all();
-    for specifier in &test_modules {
-      let file = program_state
-        .file_fetcher
-        .fetch(specifier, &mut fetch_permissions)
-        .await?;
-
-      files.push(file);
-    }
-
-    let prepare_roots = files
-      .iter()
-      .filter(|file| file.media_type != MediaType::Unknown)
-      .map(|file| file.specifier.clone())
-      .collect();
-
-    prepare_roots
-  };
+  let module_specifiers = specifiers
+    .iter()
+    .filter_map(|(specifier, mode)| {
+      if *mode != TestMode::Documentation {
+        Some(specifier.clone())
+      } else {
+        None
+      }
+    })
+    .collect();
 
   program_state
     .prepare_module_graph(
-      prepare_roots,
-      lib.clone(),
+      module_specifiers,
+      lib,
       Permissions::allow_all(),
-      permissions.clone(),
+      permissions,
       program_state.maybe_import_map.clone(),
     )
     .await?;
 
-  if no_run {
-    return Ok(());
-  }
-
-  let (sender, receiver) = channel::<TestEvent>();
-
-  let join_handles = test_modules.iter().map(move |main_module| {
-    let program_state = program_state.clone();
-    let main_module = main_module.clone();
-    let permissions = permissions.clone();
-    let filter = filter.clone();
-    let sender = sender.clone();
+  Ok(())
+}
 
-    tokio::task::spawn_blocking(move || {
-      let join_handle = std::thread::spawn(move || {
-        let future = test_specifier(
-          program_state,
-          main_module,
-          permissions,
-          filter,
-          shuffle,
-          sender,
-        );
+/// Test a collection of specifiers concurrently.
+#[allow(clippy::too_many_arguments)]
```

Comment:
This is unlucky, any ideas how we could simplify function signature?

Reply:
Yeah we can do an options struct.